src/commands/DumpRulesCommand.ts

```diff
@@ -14,10 +14,62 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { Mjolnir } from "../Mjolnir";
 import { RichReply } from "matrix-bot-sdk";
+import { Mjolnir } from "../Mjolnir";
+import { RULE_ROOM, RULE_SERVER, RULE_USER } from "../models/BanList";
 import { htmlEscape } from "../utils";
 
+/**
+ * List all of the rules that match a given entity.
+ * The reaso why you want to test against all rules and not just e.g. user or server is because
+ * there are situations where rules of different types can ban other entities e.g. server ACL can cause users to be banned.
+ * @param roomId The room the command is from.
+ * @param event The event containing the command.
+ * @param mjolnir A mjolnir to fetch the watched lists from.
+ * @param entity e.g. a user or room id.
+ * @returns When a response has been sent to the command.
+ */
+export async function execRulesMatchingCommand(roomId: string, event: any, mjolnir: Mjolnir, entity: string) {
+    let html = "";
+    let text = "";
+    for (const list of mjolnir.lists) {
+        const matches =  list.rulesMatchingEntity(entity)
+
+        if (matches.length === 0) {
+            continue;
+        }
+
+        const matchesInfo = `Found ${matches.length} ` + (matches.length === 1 ? 'match:' : 'matches:');
+        const shortcodeInfo = list.listShortcode ? ` (shortcode: ${htmlEscape(list.listShortcode)})` : '';
+
+        html += `<a href="${htmlEscape(list.roomRef)}">${htmlEscape(list.roomId)}</a>${shortcodeInfo} ${matchesInfo}<br/><ul>`;
```

Comment:
I assume that looking up room aliases here would be too slow?

Reply:
Replacing the room ids with pills in the response should already happen (in the reply code) but i will double check

Reply:
Yep